Source/WebCore/page/ContextMenuController.cpp

```diff
@@ -998,29 +1006,39 @@ void ContextMenuController::populate()
                 appendItem(StopItem, m_contextMenu.get());
                 appendItem(ReloadItem, m_contextMenu.get());
 #else
-                if (frame->page() && frame->page()->backForward().canGoBackOrForward(-1))
-                    appendItem(BackItem, m_contextMenu.get());
-
-                if (frame->page() && frame->page()->backForward().canGoBackOrForward(1))
-                    appendItem(ForwardItem, m_contextMenu.get());
-
-                // use isLoadingInAPISense rather than isLoading because Stop/Reload are
-                // intended to match WebKit's API, not WebCore's internal notion of loading status
-                if (loader.documentLoader()->isLoadingInAPISense())
-                    appendItem(StopItem, m_contextMenu.get());
-                else
-                    appendItem(ReloadItem, m_contextMenu.get());
+                if (frame->isMainFrame() || (frame->ownerElement() && !frame->ownerElement()->document().isPDFDocument())) {
+                    auto* framePage = frame->page();
```

Comment:
This variable can be called `page`, we don't use the term `framePage` elsewhere is the code.

Reply:
I added 3 new local variables: ownerElement, isPDFDocument, and isMainFrame. For areas where we checked for both ownerElement and isPDFDocument, I changed it to just check for isPDFDocument since the value of that boolean also implicitly checks for ownerElement. If you think there might be something wrong with that logic let me know and I can add back in the ownerElement check.

Reply:
I agree we should call this `page` rather than `framePage`